tests/test_validation.py

```diff
@@ -100,39 +100,11 @@ def test_object_assignment_with_nested_null():
         model.meta.object_attribute = {"string_attribute": None}
 
 
-@pytest.mark.xfail(reason="validation of a required attribute not yet implemented", strict=True)
-def test_required_attribute_assignment():
+def test_validation_for_required(tmp_path):
+    fn = tmp_path / "test.fits"
     model = RequiredModel()
-
-    with warnings.catch_warnings():
-        warnings.simplefilter("error")
-        model.meta.required_attribute = "foo"
-
-    with pytest.warns(ValidationWarning):
-        model.meta.required_attribute = None
-
-
-@pytest.mark.xfail(reason="validation of required attributes not yet implemented", strict=True)
-def test_validation_on_delete():
```

Comment:
just for my information, does any validation happen on delete of attributes?  I can see why it would be desired so that you can't remove a required attribute.  What is the history of this?

Reply:
Thanks, yeah I tend to agree